frontend/fingertips-frontend/components/forms/SearchForm/index.tsx

```diff
@@ -1,20 +1,54 @@
 'use client';
 
-import { SearchFormState, getSearchSuggestions } from './searchActions';
-import { Button, InputField, H3, Link } from 'govuk-react';
+import { SearchFormState, getArea } from './searchActions';
+import { Button, InputField, H3 } from 'govuk-react';
+import { useSearchParams, useRouter } from 'next/navigation';
 import { spacing } from '@govuk-react/lib';
 import styled from 'styled-components';
 import { GovukColours } from '@/lib/styleHelpers/colours';
+import AreaAutoCompleteInputField from '@/components/molecules/AreaAutoCompleteSearchPanel';
+import { AreaDocument } from '@/lib/search/searchTypes';
+import { SearchParams } from '@/lib/searchStateManager';
+import { useEffect, useState } from 'react';
 
 const StyledInputField = styled(InputField)(
   spacing.withWhiteSpace({ marginBottom: 6 })
 );
 
-export const SearchForm = ({
-  searchFormState,
-}: {
+interface SearchFormProps {
   searchFormState: SearchFormState;
-}) => {
+}
+
+export const SearchForm = ({ searchFormState }: SearchFormProps) => {
+  const params = useSearchParams();
+  const router = useRouter();
+  const [areaCode, setAreaCode] = useState<string>('');
+  const [defaultAreas, setDefaultAreas] = useState<AreaDocument[]>([]);
+
+  const updateUrlWithSelectedArea = (selectedAreaCode: string | undefined) => {
+    const urlParams = new URLSearchParams(params);
+    if (!selectedAreaCode) {
+      urlParams.delete(SearchParams.AreasSelected);
+    } else {
+      urlParams.set(SearchParams.AreasSelected, selectedAreaCode);
+    }
+    router.push(`?${urlParams.toString()}`, { scroll: false });
```

Comment:
This needs to be changed. We shouldn't be using the `useSearchParams()` within the component anymore. We should be pulling the params at the page level is the new pattern we are trying to follow and then passing the `searchState` as a prop to the component. 

In addition we shouldn't be directly updating the urlParams. But using the `searchStateManager` for this purpose.

If you would like to discuss this in more details give me a shout.